builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1231,6 +1325,12 @@ func (vm *virtualMachine) deployVirtualMachine(c *govmomi.Client) error {
 			return err
 		}
 	}
+
+	// Create the cdroms if needed.
+	if err := createCdroms(newVM, vm.cdroms); err != nil {
```

Comment:
Can you move this to L1310 (before the `newVM.Customize(context.TODO(), customSpec)`)?